kornia/augmentation/_2d/intensity/planckian_jitter.py

```diff
@@ -0,0 +1,118 @@
+                                          [0.3543, 0.4719, 0.6472],
+                                          [0.3480, 0.4719, 0.6555],
+                                          [0.3421, 0.4719, 0.6631],
+                                          [0.3365, 0.4719, 0.6702],
+                                          [0.3313, 0.4719, 0.6766],
+                                          [0.3263, 0.4719, 0.6826],
+                                          [0.3217, 0.4719, 0.6882]])
+                    }
+
+
+class PlanckianJitter(IntensityAugmentationBase2D):
+    r"""Apply planckian jitter transformation to input tensor
+    This is physics based color augmentation, that creates realistic
+    variations in chromaticity, this can simulate the illumination
+    changes in the scene. This is based on https://arxiv.org/pdf/2202.07993.pdf
+
+    Args:
+        mode: 'blackbody' or 'CIED'.
+        same_on_batch: apply the same transformation across the batch.
+        p: probability that the random erasing operation will be performed.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                                to the batch form (False).
+    Shape:
+    - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+    - Output: :math:`(B, C, H, W)`
+
+    Note:
+        Input tensor must be float and normalized into [0, 1]
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> input = torch.randn(1, 3, 32, 32)
+        >>> aug = PlanckianJitter(mode='CIED', p=1.0)
+        >>> aug(input)
+    """
+
+    def __init__(self,
+                 mode: str = "blackbody",
+                 select_from: Optional[Union[Tensor, int, List[int]]] = None,
+                 same_on_batch: bool = False,
+                 p: float = 0.5,
+                 keepdim: bool = False,
+                 return_transform: Optional[bool] = None) -> None:
+        super().__init__(p=p, return_transform=return_transform, same_on_batch=same_on_batch, keepdim=keepdim)
+
+        self.idx = select_from
+        if select_from:
+            self.register_buffer('pl',
+                                 planckian_coeffs[mode][select_from])
+        else:
+            self.register_buffer('pl', planckian_coeffs[mode])
+        self._param_generator = cast(rg.PlanckianJitterGenerator,
+                                     rg.PlanckianJitterGenerator([0, self.pl.shape[0]]))
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], transform: Optional[Tensor] = None
+    ) -> Tensor:
+
+        list_idx = params['idx'].tolist()
+        mult_ch_zero = self.pl[list_idx][:, 0] / self.pl[list_idx][:, 1]
+        mult_ch_two = self.pl[list_idx][:, 0] / self.pl[list_idx][:, 1]
+        input[:, 0, ...] = input[:, 0, ...] * mult_ch_zero.unsqueeze(-1).unsqueeze(-1)
+        input[:, 2, ...] = input[:, 2, ...] * mult_ch_two.unsqueeze(-1).unsqueeze(-1)
+
+        clamp_val = torch.Tensor([1.0])
```

Comment:
```suggestion
        clamp_val = torch.ones_like(input)
```
This will assure the exact device and dtype